packages/protocol/contracts/RequesterStore.sol

```diff
@@ -69,9 +79,8 @@ contract RequesterStore is IRequesterStore {
         override
         onlyRequesterAdmin(requesterIndex)
     {
-        // Initialize the client nonce if it is being endorsed for the first
-        // time for consistent request gas cost
-        if (endorsementStatus && clientAddressToNoRequests[clientAddress] == 0)
+        // Initialize the client nonce for consistent request gas cost
+        if (clientAddressToNoRequests[clientAddress] == 0)
```

Comment:
Does the `endorsementStatus` matter here?

Reply:
Previously if the initial `endorsementStatus` was `false` and it was being updated as `false`, the nonce wasn't being initialized. I figured `endorsementStatus` is completely unrelated to the nonce initialization so it should be initialized anyway.

Here one view is that `updateClientEndorsementStatus` should revert if the `endorsementStatus` is the same with the current endorsement status (because it wouldn't be "updating" the status). Then this wouldn't be an issue (`endorsementStatus` would always be `true` for the first non-reverting call) but I'm not a fan of reverting that much (though maybe the function should be renamed as `setClientEndorsementStatus`).

Reply:
> I figured endorsementStatus is completely unrelated to the nonce initialization so it should be initialized anyway.
This is what I thought and it seems better without the additional check. Just wanted to check

> maybe the function should be renamed as `setClientEndorsementStatus`
I like this name more

Reply:
What do you think about `setProvider` vs. `createProvider`, since that's the same thing

Reply:
I like `setProviderParameters` - seems pretty clear in what it's doing.

Calling `createProvider` multiple times implies that it creates multiple providers, when in fact it's updating/overwriting on subsequent calls